pkg/remoteclusters/gateway_address.go

```diff
@@ -19,37 +19,39 @@ package remoteclusters
 import (
 	"context"
 
-	corev1 "k8s.io/api/core/v1"
+	"github.com/pkg/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 
 	istiov1beta1 "github.com/banzaicloud/istio-operator/pkg/apis/istio/v1beta1"
-	"github.com/banzaicloud/istio-operator/pkg/k8sutil"
+	"github.com/banzaicloud/istio-operator/pkg/resources/ingressgateway"
 	"github.com/banzaicloud/istio-operator/pkg/util"
 )
 
-func (c *Cluster) getIngressGatewayAddress(remoteIstio *istiov1beta1.RemoteIstio, istio *istiov1beta1.Istio) error {
-	if !util.PointerToBool(istio.Spec.MeshExpansion) {
+func (c *Cluster) SetIngressGatewayAddress(remoteIstio *istiov1beta1.RemoteIstio, istio *istiov1beta1.Istio) error {
```

Comment:
Could the `func (r *ReconcileConfig) setIngressGatewayAddress(istio *istiov1beta1.Istio, logger logr.Logger) ([]string, error)` func be reused here? Feels like there is some code duplication here which is already in that function.